packages/pwa-kit-extension-sdk/src/react/classes/ApplicationExtension.ts

```diff
@@ -39,12 +39,12 @@ export class ApplicationExtension<
     constructor(config: Config) {
         super(config)
 
-        if (this.getRoutesAsync) {
+        if (!isServerSide()) {
```

Comment:
I think a bit more verbose here is helpful. In the constructor:
 - deserialize on the client side
 - monkey-patch `getRoutesAsync` to return cached value ← both on server and client sides